numpy/core/src/multiarray/compiled_base.dispatch.c

```diff
@@ -0,0 +1,104 @@
+/**
+ * @targets $maxopt baseline
+ * SSE2 AVX2
+ * VSX VSX2
+ * NEON ASIMDDP
+ */
+#include "compiled_base.h"
+
+/*
+ * This function packs boolean values in the input array into the bits of a
+ * byte array. Truth values are determined as usual: 0 is false, everything
+ * else is true.
+ */
+NPY_NO_EXPORT void NPY_CPU_DISPATCH_CURFX(compiled_base_pack_inner)
+(const char *inptr, npy_intp element_size,  npy_intp n_in, npy_intp in_stride, char *outptr, npy_intp n_out, npy_intp out_stride, char order)
+{
+    /*
+     * Loop through the elements of inptr.
+     * Determine whether or not it is nonzero.
+     *  Yes: set corresponding bit (and adjust build value)
+     *  No:  move on
+     * Every 8th value, set the value of build and increment the outptr
+     */
+    npy_intp index = 0;
+    int remain = n_in % 8;              /* uneven bits */
+
+#if defined(NPY_HAVE_SSE) || defined(NPY_HAVE_AVX2) || defined(NPY_HAVE_NEON)
```

Comment:
Shouldn't this be a more general condition to get VSX and any possible future platforms as well?

Reply:
yes, I think VSX can benefit from this optimize too, but I don't have any PowerPC device for testing. The final macro should be `NPY_SIMD`.

Reply:
So please use `NPY_SIMD`. We have ppc64le CI testing, it should smoke out problems.

Reply:
changed to `NPY_SIMD`